### PR TITLE
migrate helm release version secrets to helm namespace

### PR DESCRIPTION
**Pull request by pavansokkenagaraj**
#### What this PR does / why we need it:
Helm upgrade fails for charts installed with KOTS versions earlier than 1.95.0

Helm Release Version Secrets were created in kots installed namespace earlier to kots v1.95.0
and in kots v1.95.0 after secrets started to respect kots HelmChart namespace, 
helm upgrade expects these secrets to be their respective namespace

#### Which issue(s) this PR fixes:

Fixes # https://app.shortcut.com/replicated/story/72389/helm-upgrade-fails-for-charts-installed-with-kots-versions-earlier-than-1-95-0

#### Does this PR introduce a user-facing change?

```release-note
fix a bug where helm release secret was created in kotsadm namespace instead helm release namespace
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

**Review by cbodonnell — DISMISSED** [review]

File: pkg/helm/update_release.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	}
+
+	// set release namespace to the new namespace
+	release.Namespace = releaseNamespace
+	releaseStr, err := encodeRelease(release)
+	if err != nil {
+		return errors.Wrapf(err, "failed to encode release")
+	}
+
+	newReleaseSecret := corev1.Secret{
```
Comment (sgalsaleh) [inline pkg/helm/update_release.go:75]: Hmm, any reason to not just use the existing secret? Who knows what Helm will add to it later. That also seems to be how this article did it https://blog.random.io/moving-helm-release-to-another-kubernetes-namespace/

File: pkg/helm/update_release.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		}
+		return errors.Wrapf(err, "failed to list release secrets for %s", releaseName)
+	}
+
+	if len(secretList.Items) == 0 {
+		return nil
+	}
+
+	for _, secret := range secretList.Items {
+		if secret.Namespace != releaseNamespace {
```
Comment (sgalsaleh) [inline pkg/helm/update_release.go:51]: this if statement is not needed, no? if you really want to have the function to be a no-op if the two namespaces match, i'd suggest you check for this at the beginning of the function and just return?

File: pkg/operator/client/deploy.go [hunk 3]
```diff
@@ -522,11 +524,27 @@ func (c *Client) installWithHelm(helmDir string, targetNamespace string, kotsCha
 		installDir := filepath.Join(chartsDir, dir.Name)
 		args := []string{"upgrade", "-i", dir.ReleaseName, installDir, "--timeout", "3600s"}
 
+		upgradeNamespace := "default"
```
Comment (sgalsaleh) [inline pkg/operator/client/deploy.go:527]: too many variables that refer to a namespace which is pretty confusing. my recommendation is:

- pass the `targetNamespace` to the `getSortedCharts` function and if the `dir.Namespace` is empty, set it to the `targetNamespace` (if it's not `.`.
- change the if statement below to just be `if dir.Namespace != util.AppNamespace()`. e.g.: let's not define 2 more variables that refer to a namespace

Comment (sgalsaleh) [reply pkg/operator/client/deploy.go:527]: from what i can see and know, it should never be `.` when you get to this the `installWithHelm` function, so you don't need to check for that

**Pull request merged**